### ISSUE TITLE
Unable to install pytorch-scatter

### ISSUE THREAD

**Issue by bymbhaskar** [rusty1s/pytorch_scatter#195]
I am getting following error while installing pytorch-scatter:

bhaskar@bhaskar-Precision-5820-Tower:~$ pip3 install torch-scatter
WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
Defaulting to user installation because normal site-packages is not writeable
Collecting torch-scatter
  Using cached torch_scatter-2.0.5.tar.gz (20 kB)
Building wheels for collected packages: torch-scatter
  Building wheel for torch-scatter (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"'; __file__='"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-pu8jgwcw
       cwd: /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/
  Complete output (34 lines):
  /home/bhaskar/.local/lib/python3.6/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 9000). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
    return torch._C._cuda_getDeviceCount() > 0
  No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda-10.2'
  /usr/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'python_requires'
    warnings.warn(msg)
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.6/torch_scatter
  creating build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  running build_ext
  building 'torch_scatter._segment_coo' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/tmp
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/cpu
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/TH -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/THC -I/usr/include/python3.6m -c /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.cpp -o build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.o -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE="_gcc" -DPYBIND11_STDLIB="_libstdcpp" -DPYBIND11_BUILD_ABI="_cxxabi1011" -DTORCH_EXTENSION_NAME=_segment_coo -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
  /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.cpp:1:20: fatal error: Python.h: No such file or directory
  compilation terminated.
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
Failed to build torch-scatter
Installing collected packages: torch-scatter
    Running setup.py install for torch-scatter ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"'; __file__='"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-sccodl2o/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/bhaskar/.local/include/python3.6m/torch-scatter
         cwd: /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/
    Complete output (34 lines):
    /home/bhaskar/.local/lib/python3.6/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 9000). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at  /pytorch/c10/cuda/CUDAFunctions.cpp:100.)
      return torch._C._cuda_getDeviceCount() > 0
    No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda-10.2'
    /usr/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.6/torch_scatter
    copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.6/torch_scatter
    creating build/lib.linux-x86_64-3.6/torch_scatter/composite
    copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
    copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
    copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
    copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
    running build_ext
    building 'torch_scatter._segment_coo' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/tmp
    creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi
    creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7
    creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc
    creating build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/cpu
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/torch/csrc/api/include -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/TH -I/home/bhaskar/.local/lib/python3.6/site-packages/torch/include/THC -I/usr/include/python3.6m -c /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.cpp -o build/temp.linux-x86_64-3.6/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.o -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE="_gcc" -DPYBIND11_STDLIB="_libstdcpp" -DPYBIND11_BUILD_ABI="_cxxabi1011" -DTORCH_EXTENSION_NAME=_segment_coo -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    /tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/csrc/segment_coo.cpp:1:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"'; __file__='"'"'/tmp/pip-install-rybgdfbi/torch-scatter_ef0aa882c9d545f395e479941eb600e7/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-sccodl2o/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/bhaskar/.local/include/python3.6m/torch-scatter Check the logs for full command output.

**Comment by rusty1s** [comment]
This might be related: https://github.com/pytorch/pytorch/issues/18387
